### DISCUSSION TITLE
How do I convert *large* files?

### DISCUSSION THREAD

**Discussion by aflyingpumpkin** [ImageMagick/ImageMagick#5243] [Help]
I am trying to resize an 11Gb tiff file using ImageMagick though I end up with the error `unable to write pixel cache`...`No space left on device`. Does anyone know how to get around this?
Full error:
```
convert: Unknown field with tag 33550 (0x830e) encountered. `TIFFReadDirectory' @ warning/tiff.c/TIFFWarnings/956.
convert: Unknown field with tag 33922 (0x8482) encountered. `TIFFReadDirectory' @ warning/tiff.c/TIFFWarnings/956.
convert: Unknown field with tag 34735 (0x87af) encountered. `TIFFReadDirectory' @ warning/tiff.c/TIFFWarnings/956.
convert: Unknown field with tag 34736 (0x87b0) encountered. `TIFFReadDirectory' @ warning/tiff.c/TIFFWarnings/956.
convert: Unknown field with tag 34737 (0x87b1) encountered. `TIFFReadDirectory' @ warning/tiff.c/TIFFWarnings/956.
convert: Unknown field with tag 42113 (0xa481) encountered. `TIFFReadDirectory' @ warning/tiff.c/TIFFWarnings/956.
convert: unable to write pixel cache '/tmp/magick-HhWvV-H8gkeiqtWxCuKaCUUQaf2ALKS3': No space left on device @ error/cache.c/WritePixelCachePixels/5850.
convert: no images defined `Mars_HeightMap_1000.tiff' @ error/convert.c/ConvertImageCommand/3325.
```

I tried running `mount -t tmpfs tmpfs /tmp -o size=15G,mode=1777,remount` to increase the size of `/tmp` though still had no luck. Does anyone have an idea on how to get around this?

**Comment by fmw42** [comment]
Most of those are warnings that tiff tags exist that IM does no know about. You can remove those by adding -quiet to your command line. The main issue is that your /tmp directory is full or filling. How big are your images? Are you aborting the processing. That may leave large tmp files from Imagemagick that will not be automatically removed. Delete manually any Imagemagick tmp files. If that does not work, then set up your Imagemagick environment variable to specify a different directory for your tmp directory. See https://imagemagick.org/script/resources.php#environment

**Comment by urban-warrior** [comment]
See https://imagemagick.org/script/architecture.php#tera-pixel for tips on how to properly deal with huge images.
